### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Kalixone**

### CODE REVIEW

File: src/main/java/core/basesyntax/service/InvalidUserException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.service;
+
+public class InvalidUserException extends RuntimeException {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/service/InvalidUserException.java:3]:
```suggestion
public class RegistrationException extends RuntimeException {
```

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+
+    private void validateUser(User user) {
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidUserException("Invalid login - min 6 characters");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidUserException("Invalid password - min 6 characters");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new InvalidUserException("Invalid age - min 18 years old");
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:27]: image

File: src/test/java/core/basesyntax/dao/StorageDaoImplTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class StorageDaoImplTest {
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/dao/StorageDaoImplTest.java:11]: delete this class - it is out of scope
image

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:18]: we do not need StorageDao in this class - remove it

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+        user.setAge(25);
+
+        User addedUser = storageDao.add(user);
+
+        assertEquals(user, addedUser);
+        assertNotNull(addedUser.getId());
+    }
+
+    @Test
+    void shouldThrowExceptionForEmptyLogin() {
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:41]: image

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java (hunk 4 above)
Comment (Kalixone) [reply src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:18]: Could you explain me why, we should delete StorageDao from this class?, and how to do it without damaging it?

File: src/main/java/core/basesyntax/service/InvalidUserException.java (hunk 1 above)
Comment (Rommelua) [reply src/main/java/core/basesyntax/service/InvalidUserException.java:3]: not fixed - rename this class please

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java (hunk 2 above)
Comment (Rommelua) [reply src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:27]: not fixed

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java (hunk 4 above)
Comment (Rommelua) [reply src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:18]: Because we do not need it. Give at least one reason why we need it here?

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+
+        RegistrationException invalidUserException = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Invalid login - min 6 characters", invalidUserException.getMessage());
+    }
+
+    @Test
+    void register_edgeCaseLogin_notOk() {
+        User user = new User();
+        user.setLogin("1");
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:75]: not ok edge case for login is length = 5

**Review by Rommelua — APPROVED** [review]
